webpack/assets/javascripts/react_app/routes/index.js

```diff
@@ -1,23 +1,22 @@
 import React from 'react';
-import PropTypes from 'prop-types';
 import { Switch, Route } from 'react-router-dom';
-
 import { routes } from './routes';
 
 let currentLocation = null;
 
-const AppSwitcher = () => (
+const AppSwitcher = serverProps => (
   <Switch>
-    {routes.map(({ render, ...props }) => (
+    {routes.map(({ render, path, ...routeProps }) => (
       <Route
-        {...props}
-        key={props.path}
+        path={path}
+        key={path}
+        {...routeProps}
         render={renderProps => {
           const railsContainer = document.getElementById('rails-app-content');
           if (railsContainer) railsContainer.remove();
           currentLocation = renderProps.location;
 
-          return render(renderProps);
+          return render({ ...renderProps, ...serverProps });
```

Comment:
Does it mean only routes will receive the `metadata` props?

imo it should be an `application-metadata` reducer instead so all components get the app-metadata info by using selectors.